compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -490,6 +490,15 @@ setMatmulVectorDistributionConfig(IREE::GPU::TargetAttr target,
   SmallVector<int64_t, 4> bounds = op.getStaticLoopRanges();
   FailureOr<mlir::linalg::ContractionDimensions> contractionDims =
       mlir::linalg::inferContractionDims(op);
+  if (failed(contractionDims)) {
+    assert(op->getNumResults() > 1 &&
+           "expected horizontally fused contraction op");
```

Comment:
Assert condition looks imprecise; `IREE::LinalgExt::isaHorizontallyFusedContraction(linalgOp)` and/or `op.getNumDpsInputOperands() > 2`?